src/backend/distributed/master/master_modify_multiple_shards.c

```diff
@@ -312,3 +316,106 @@ ShouldExecuteTruncateStmtSequential(TruncateStmt *command)
 
 	return false;
 }
+
+
+/*
+ * lock_relation_if_exists gets a relation name and lock mode
+ * and returns true if the relation exists and can be locked with
+ * the given lock mode. If the relation doesn't exists, the function
+ * return false.
+ *
+ * The relation name should be qualified with the schema name.
+ *
+ * The function errors out of the lockmode isn't defined in the PostgreSQL's
+ * explicit locking table.
+ */
+Datum
+lock_relation_if_exists(PG_FUNCTION_ARGS)
+{
+	text *relationName = PG_GETARG_TEXT_P(0);
+	text *lockModeText = PG_GETARG_TEXT_P(1);
+
+	Oid relationId = InvalidOid;
+	char *lockModeCString = text_to_cstring(lockModeText);
+	List *relationNameList = NIL;
+	RangeVar *relation = NULL;
+	LOCKMODE lockMode = NoLock;
+
+	/* ensure that we're in a transaction block */
+	RequireTransactionChain(true, "lock_relation_if_exists");
```

Comment:
`RequireTransactionChain` is replaced with `RequireTransactionBlock` in PG11.

Please use  `RequireTransactionBlock` in code, and add

 ```c
#define RequireTransactionBlock RequireTransactionChain
```

in `version_compat.h` for pre-11 builds

Reply:
Thanks, I'll also switch using pg11 for my tests